src/config/index.ts

```diff
@@ -1,20 +1,19 @@
 import fs, { readFileSync } from 'node:fs';
 import { join } from 'node:path';
 
-import { go } from '@api3/promise-utils';
+import { goSync } from '@api3/promise-utils';
 import dotenv from 'dotenv';
 
 import { configSchema } from './schema';
 import { interpolateSecrets, parseSecrets } from './utils';
 
-export const loadConfig = async () => {
-  const configPath = join(__dirname, '../../config');
+export const loadConfig = (configPath = join(__dirname, '../../config'), airseekerConfigPath = 'airseeker.json') => {
   const rawSecrets = dotenv.parse(readFileSync(join(configPath, 'secrets.env'), 'utf8'));
 
-  const goLoadConfig = await go(async () => {
-    const rawConfig = JSON.parse(fs.readFileSync(join(configPath, 'airseeker.json'), 'utf8'));
+  const goLoadConfig = goSync(() => {
+    const rawConfig = JSON.parse(fs.readFileSync(join(configPath, airseekerConfigPath), 'utf8'));
     const secrets = parseSecrets(rawSecrets);
-    return configSchema.parseAsync(interpolateSecrets(rawConfig, secrets));
+    return configSchema.safeParse(interpolateSecrets(rawConfig, secrets));
```

Comment:
Good catch. The config does not need to be parsed asynchronously.

However, you should use the `.parse` instead of `.safeParse` here (the response type is wrong now).

Reply:
Weirdly it didn't fail, I've swapped it out 👌